src/tracker/tracker_sampler.jl

```diff
@@ -5,94 +5,97 @@ mutable struct BoxROI{T <: AbstractArray, S <: AbstractArray} <: AbstractROI
 end
 
 abstract type AbstractTrackerSampler end
-mutable struct CurrentSample{F <: Float64, S <: Symbol} <: AbstractTrackerSampler
+mutable struct CurrentSampler{F <: AbstractFloat} <: AbstractTrackerSampler
     overlap::F
     search_factor::F
-    mode::S
-    sampling_ROI::SVector{4, Integer}
-    valid_ROI::SVector{4, Integer}
+    mode::S where S <: Symbol
+    sampling_region::SVector{4, Integer}
+    valid_region::SVector{4, Integer}
     tracked_patch::SVector{4, Integer}
-    CurrentSample{F, S}(overlap, search_factor, mode) where{F <: Float64, S <: Symbol} = new(overlap, search_factor, mode)
+    CurrentSampler{F}(overlap, search_factor) where{F <: AbstractFloat} = new(overlap, search_factor)
```

Comment:
style tip: generally, we add one space after `where`, e.g.,
`CurrentSampler{F}(overlap, search_factor) where {F <: AbstractFloat} = new(overlap, search_factor)`
or
`CurrentSampler{F}(overlap, search_factor) where F <: AbstractFloat = new(overlap, search_factor)`